src/microbe_stage/CompoundCloudSystem.cs

```diff
@@ -271,19 +268,25 @@ public void GetAllAvailableAt(Vector3 worldPosition, Dictionary<Compound, float>
                     if (x < 0 || y < 0)
                         continue;
 
-                    // Circle check
-                    if (Mathf.Pow(x - cloudRelativeX, 2) + Mathf.Pow(y - cloudRelativeY, 2) >
-                        localGrabRadiusSquared)
-                    {
-                        // Not in it
+                    // Delta calculation using simple math to calculate the square distance
+                    float deltaX = x - cloudRelativeX;
+                    float deltaY = y - cloudRelativeY;
+                    float distanceSquared = deltaX * deltaX + deltaY * deltaY;
+
+                    // This ensures that the distance will be non zero and scales it down
+                    // Despite its simplicity it helps not getting negative values and
+                    // keeps the same results as the old one
+                    float distance = (distanceSquared + 1.0f) / 2.0f;
```

Comment:
I don't really understand how this math goes from squared to non-squared distance coordinates.

Reply:
The naming of the variable i've used could be a bit misleading it should have probably been named in a differtent manner like "distanceaprox" or similar which is clearer in my opinion, however most of the logic that ensures the cloud's round shape precedes these lines. I appreciate and  understand your decision and i'm eager to share some benchmark results in here or dms to help with the final decision.